service/search/es.go

```diff
@@ -161,118 +174,205 @@ var esSetting = m{
 
 // NewESEngine Elasticsearch検索エンジンを生成します
 func NewESEngine(mm message.Manager, cm channel.Manager, repo repository.Repository, logger *zap.Logger, config ESEngineConfig) (Engine, error) {
-	// es接続
-	client, err := elastic.NewClient(elastic.SetURL(config.URL), elastic.SetSniff(false))
+	// esクライアント作成
+	client, err := elasticsearch.NewClient(elasticsearch.Config{
+		Addresses: config.URL,
+	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to init search engine: %w", err)
+		return nil, fmt.Errorf("failed to init Elasticsearch: %w", err)
 	}
 
 	// esバージョン確認
-	version, err := client.ElasticsearchVersion(config.URL)
+	infoRes, err := client.Info()
+	defer infoRes.Body.Close()
 	if err != nil {
-		return nil, fmt.Errorf("failed to fetch es version: %w", err)
+		return nil, fmt.Errorf("failed to get Elasticsearch info: %w", err)
+	}
+	if infoRes.IsError() {
+		return nil, fmt.Errorf("failed to get Elasticsearch info: %s", infoRes.String())
+	}
+
+	var r map[string]interface{}
```

Comment:
こここうすれば後ろの型キャスト排除してJSONのデコードで全部解決できる説ある
```suggestion
	var r struct {
		Version struct {
			Number string `json:"number"`
		} `json:"version"`
	}
```